src/realm/object-store/impl/collection_notifier.cpp

```diff
@@ -51,133 +51,74 @@ CollectionNotifier::get_modification_checker(TransactionChangeInfo const& info,
         };
     }
 
+    // If the table in question has no outgoing links it will be the only entry in `m_related_tables`.
+    // In this case we do not need a `DeepChangeChecker` and check the modifications against the
+    // `ObjectChangeSet` within the `TransactionChangeInfo` for this table directly.
     if (m_related_tables.size() == 1) {
-        auto& object_set = info.tables.find(m_related_tables[0].table_key.value)->second;
+        auto& object_chenge_set = info.tables.find(m_related_tables[0].table_key.value)->second;
         return [&](ObjectChangeSet::ObjectKeyType object_key) {
-            return object_set.modifications_contains(object_key);
+            if (all_callbacks_filtered()) {
+                return object_chenge_set.modifications_contains(object_key, get_filtered_column_keys(true));
```

Comment:
We don't want to be building this vector for every single row we're checking. This is part of the hot inner loop where we're iterating over a potentially very large number of rows.

Reply:
I've removed the check from the lambda in 667011b.
An alternative idea would be to calculate it every time the related tables are recalculated and save it for later.